xclim/indices/run_length.py

```diff
@@ -121,10 +128,13 @@ def rle(
 
     # back fill nans with first position after
     z = b.bfill(dim=dim)
+
     # calculate lengths
     d = z.diff(dim=dim) - 1
     d = d.where(d >= 0)
```

Comment:
If there is no significant performance penalty from the reindexing operation, I'm happy with the proposed solution. If there is, then I think you'd need to something like
```
d = d.where(d <= 0)
```
when `lastday is True`.

Reply:
It doesn't look like the reindexing is impacting perfomance very much, although I've only measured the tests:

```
xclim/testing/tests/test_run_length.py::test_rle[False-True-True] lastday: True
0.0477299690246582
PASSED
xclim/testing/tests/test_run_length.py::test_rle[False-True-False] lastday: True
0.02160477638244629
PASSED
xclim/testing/tests/test_run_length.py::test_rle[False-False-True] lastday: False
0.04825186729431152
PASSED
xclim/testing/tests/test_run_length.py::test_rle[False-False-False] lastday: False
0.01996779441833496
```

Reply:
Excellent then.